xrmocap/ops/triangulation/point_selection/reprojection_error_point_selector.py

```diff
@@ -0,0 +1,162 @@
+                An ndarray or a nested list of points2d, in shape
+                [n_view, n_keypoints, 2+n], n >= 0.
+            init_points_mask (Union[np.ndarray, list, tuple], optional):
+                An ndarray or a nested list of mask, in shape
+                [n_view, n_keypoints, 1].
+                If points_mask[index] == 1, points[index] is valid
+                for triangulation, else it is ignored.
+                If points_mask[index] == np.nan, the whole pair will
+                be ignored and not counted by any method.
+                Defaults to None.
+
+        Returns:
+            np.ndarray:
+                An ndarray or a nested list of mask, in shape
+                [n_view, n_keypoints, 1].
+        """
+        points, init_points_mask = prepare_triangulate_input(
+            camera_number=len(points),
+            points=points,
+            points_mask=init_points_mask,
+            logger=self.logger)
+        n_point = points.shape[-2]
+        n_view = points.shape[0]
+        points2d_mask = init_points_mask.copy()
+        for point_idx in range(n_point):
+            selected_view = self.get_view_index(
+                point=points[:, point_idx],
+                init_point_mask=init_points_mask[:, point_idx])
+            for view_index in range(n_view):
+                if view_index not in selected_view:
+                    points2d_mask[view_index, point_idx] = 0
+        points_mask_shape = points2d_mask.shape
+        # log stats
+        if self.verbose:
+            _, stats_table = get_valid_views_stats(
+                points2d_mask.reshape(n_view, -1, 1))
+            self.logger.info(stats_table)
+        points2d_mask = points2d_mask.reshape(*points_mask_shape)
+        return points2d_mask
+
+    def get_view_index(self, point: np.ndarray,
+                       init_point_mask: np.ndarray) -> list:
+        """Get a list of camera indexes. This selector will loop triangulate
+        points, disable the one camera with largest reprojection error, and
+        loop again until there are self.target_camera_number left.
+
+        Args:
+            point (np.ndarray):
+                An ndarray of points2d, in shape
+                [n_view, 2+n], n >= 0.
+            init_point_mask (np.ndarray):
+                An ndarray of mask, in shape [n_view, 1].
+                If points_mask[index] == 1, points[index] is valid
+                for triangulation, else it is ignored.
+                If points_mask[index] == np.nan, the whole pair will
+                be ignored and not counted by any method.
+                Defaults to None.
+
+        Returns:
+            list:
+                A list of sorted camera indexes,
+                length == self.target_camera_number.
+        """
+        # backup shape
+        n_view = init_point_mask.shape[0]
+        # check if there's potential to search
+        remain_view = np.array([x for x in range(n_view)])
```

Comment:
What if there's already some zeros in mask?
Say `init_point_mask = np.array([0, 0, 1, 0, 1])`.
It should be something like `remain_view = np.where(init_point_mask==1)[0]`.